### PR TITLE
Fix IMapResolverTest [HZ-2349]

### LINKED ISSUE #24336: com.hazelcast.sql.impl.schema.IMapResolverTest.testCompactSerialization [HZ-2349]
_master_ (commit bdd14813949602290d6364d1ad9862194e740462)

Failed on Windows - MicrosoftJDK11: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-Windows-MicrosoftJDK11/62/testReport/com.hazelcast.sql.impl.schema/IMapResolverTest/testCompactSerialization/
<details><summary>Stacktrace:</summary>

```
java.lang.NullPointerException
	at org.assertj.core.internal.Strings.normalizeNewlines(Strings.java:685)
	at org.assertj.core.internal.Strings.assertIsEqualToNormalizingNewlines(Strings.java:677)
	at org.assertj.core.api.AbstractCharSequenceAssert.isEqualToNormalizingNewlines(AbstractCharSequenceAssert.java:1796)
	at com.hazelcast.sql.impl.schema.IMapResolverTest.testCompactSerialization(IMapResolverTest.java:81)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:829)

```
</details>

<details><summary>Standard output:</summary>

```
Finished Running Test: smokeTest in 0.037 seconds.
23:23:21,392 DEBUG || - [JobRepository] hz.IMapResolverTest_priceless_ellis.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.3.0-SNAPSHOT] Job cleanup took 0ms
23:23:21,399  INFO |testCompactSerialization| - [SqlTestSupport] Thread-1228 - Removing 0 cached plans in SqlTestSupport.@After
23:23:21,399  INFO |testCompactSerialization| - [SimpleTestInClusterSupport] Thread-1228 - Ditching 0 jobs in SimpleTestInClusterSupport.@After: []
23:23:21,399  INFO |testCompactSerialization| - [SimpleTestInClusterSupport] Thread-1228 - Destroying 0 distributed objects in SimpleTestInClusterSupport.@After: []
BuildInfo right after testCompactSerialization(com.hazelcast.sql.impl.schema.IMapResolverTest): BuildInfo{version='5.3.0-SNAPSHOT', build='20230420', buildNumber=20230420, revision=bdd1481, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'testCompactSerialization(com.hazelcast.sql.impl.schema.IMapResolverTest):'
23:23:15, accumulated pauses: 2735 ms, max pause: 1537 ms, pauses over 1000 ms: 1


No metrics recorded during the test

```
</details>

Standard output can be found here - https://s3.console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-master-Windows-MicrosoftJDK11/62/

### LINKED ISSUE #24411: com.hazelcast.sql.impl.schema.IMapResolverTest.smokeTest
_master_ (commit 86db568380dcc4265fed490d2ed6d206895ee535)

Failed on openj9-17: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-Openj9-17/63/testReport/com.hazelcast.sql.impl.schema/IMapResolverTest/smokeTest/
<details><summary>Stacktrace:</summary>

```
java.lang.NullPointerException: Cannot invoke "java.lang.CharSequence.toString()" because "actual" is null
	at org.assertj.core.internal.Strings.normalizeNewlines(Strings.java:685)
	at org.assertj.core.internal.Strings.assertIsEqualToNormalizingNewlines(Strings.java:677)
	at org.assertj.core.api.AbstractCharSequenceAssert.isEqualToNormalizingNewlines(AbstractCharSequenceAssert.java:1796)
	at com.hazelcast.sql.impl.schema.IMapResolverTest.smokeTest(IMapResolverTest.java:63)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:858)

```
</details>

<details><summary>Standard output:</summary>

```
Finished Running Test: testCompactSerialization in 0.026 seconds.
08:23:32,531  INFO |smokeTest| - [SqlTestSupport] Thread-1367 - Removing 0 cached plans in SqlTestSupport.@After
08:23:32,531  INFO |smokeTest| - [SimpleTestInClusterSupport] Thread-1367 - Ditching 0 jobs in SimpleTestInClusterSupport.@After: []
08:23:32,531  INFO |smokeTest| - [SimpleTestInClusterSupport] Thread-1367 - Destroying 0 distributed objects in SimpleTestInClusterSupport.@After: []
BuildInfo right after smokeTest(com.hazelcast.sql.impl.schema.IMapResolverTest): BuildInfo{version='5.3.0-SNAPSHOT', build='20230427', buildNumber=20230427, revision=86db568, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'smokeTest(com.hazelcast.sql.impl.schema.IMapResolverTest):'
08:23:30, accumulated pauses: 36 ms, max pause: 13 ms, pauses over 1000 ms: 0


No metrics recorded during the test

```
</details>

Standard output can be found here - https://s3.console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-master-Openj9-17/63/

### PR DESCRIPTION
**Pull request by burakgok**
The tests in `IMapResolverTest` must be run serially because `SimpleTestInClusterSupport.supportAfter()` removes all maps in the end of a test.

Fixes #24336
Fixes #24411

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible

### CODE REVIEW

File: hazelcast-sql/src/test/java/com/hazelcast/sql/impl/schema/IMapResolverTest.java [hunk 1]
```diff
@@ -20,20 +20,13 @@
...
 import org.junit.Test;
-import org.junit.experimental.categories.Category;
-import org.junit.runner.RunWith;
 
 import java.math.BigDecimal;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
-@RunWith(HazelcastParallelClassRunner.class)
-@Category({QuickTest.class, ParallelJVMTest.class})
```
Comment (Fly-Style) [inline hazelcast-sql/src/test/java/com/hazelcast/sql/impl/schema/IMapResolverTest.java:36]: We should have at least `@Category(QuickTest.class)` here. It is used for out PR builder and test coverage reports.

Comment (Fly-Style) [reply hazelcast-sql/src/test/java/com/hazelcast/sql/impl/schema/IMapResolverTest.java:36]: Also we have `HazelcastSerialClassRunner`, so, you can safely replace `HazelcastParallelClassRunner`

**Review by Fly-Style — APPROVED** [review]

**Review by Fly-Style — CHANGES_REQUESTED** [review]
We should have at least @Category(QuickTest.class) here. It is used for our PR builder and test coverage reports.

(Sorry, I misclicked)

File: hazelcast-sql/src/test/java/com/hazelcast/sql/impl/schema/IMapResolverTest.java (hunk 1 above)
Comment (burakgok) [reply hazelcast-sql/src/test/java/com/hazelcast/sql/impl/schema/IMapResolverTest.java:36]: `@Category` is inherited, so the PR builder will pick it up.

**Review by Fly-Style — APPROVED** [review]

**Review by ivanthescientist — APPROVED** [review]

**Pull request merged**